module/icp/algs/edonr/edonr.c

```diff
@@ -348,7 +348,7 @@ Q256(size_t bitlen, const uint32_t *data, uint32_t *restrict p)
  * which only goes over it by a hair (1248 bytes on ARM32).
  */
 #include <sys/isa_defs.h>	/* for _ILP32 */
-#ifdef _ILP32   /* We're 32-bit, assume small stack frames */
+#if defined (_ILP32) || defined (APPLE)   /* assume small stack frames */
```

Comment:
Let's make this:

```
#if defined (_ILP32) ||  (defined(__APPLE__) && !defined(__OPTIMIZE__))
```

(and similarly in sha2.c below)

since with optimization there is no stack warning, and since we use \_\_APPLE\_\_ canonically, not APPLE.

```
#include <stdio.h>

int
main(int argc, char *argv[])
{
        char buf[] =
#if defined(_ILP32) || (defined(__APPLE__) && !defined(__OPTIMIZE__))
            "stack fix needed";
#else
            "no stack fix needed";
#endif

        printf("%s\n", buf);
}
```

Make this o.c

```
 ⚓  smd@cla  /p/tmp  clang -o o0 o.c
 ⚓  smd@cla  /p/tmp  ./o0
stack fix needed
 ⚓  smd@cla  /p/tmp  clang -O -o o1 o.c
 ⚓  smd@cla  /p/tmp  ./o1
no stack fix needed
```